### PR TITLE
Override == method for CommandBuilder class to make it more test friendly

### PR DESCRIPTION
**Pull request by jduan**
"bundle exec rake" passed

### CODE REVIEW

File: lib/appear/util/command_builder.rb [hunk 1]
```diff
@@ -132,6 +132,17 @@ def dup
...
       end
 
+      # Override the == method
+      def ==(other)
+        other.class == self.class && other.state == self.state
+      end
+
+      # Return all the instance variables in an array. This method is used
+      # by the == method.
+      def state
```
Comment (justjake) [inline lib/appear/util/command_builder.rb]: make this `protected` please

File: spec/command_builder_spec.rb [hunk 2]
```diff
@@ -47,4 +47,12 @@
       expect(subject.to_s).to eq('foo --parent yep bar --child cow 1 2')
     end
   end
+
+  describe '#==' do
+    it 'overrides the == method properly' do
```
Comment (justjake) [inline spec/command_builder_spec.rb]: better name for this test would be `returns true for equal commands`

**Review by justjake — CHANGES_REQUESTED** [review]

File: lib/appear/util/command_builder.rb (hunk 1 above)
Comment (jduan) [reply lib/appear/util/command_builder.rb]: done

File: spec/command_builder_spec.rb (hunk 2 above)
Comment (jduan) [reply spec/command_builder_spec.rb]: That's a better name!

**Review by justjake — APPROVED** [review]

**Comment by justjake** [comment]
As the Linux builds have passed, I'm going to merge this change and ship a new version to Rubygems.org

**Pull request merged**